Realm/Tests/MigrationTests.mm

```diff
@@ -143,6 +144,40 @@ - (void)testPerRealmMigration {
     XCTAssertTrue(migrationComplete);
 }
 
+- (void)testRemovingSubclass {
+    @autoreleasepool {
+        RLMObjectSchema *objectSchema = [[RLMObjectSchema alloc] initWithClassName:@"DeletedClass" objectClass:RLMObject.class properties:@[]];
+        RLMRealm *realm = [self realmWithSingleObject:objectSchema];
+
+        [realm transactionWithBlock:^{
+            [realm createObject:@"DeletedClass" withValue:@[]];
+        }];
+    }
+
+    @autoreleasepool {
+        // apply migration
+        [RLMRealm setSchemaVersion:1 forRealmAtPath:RLMTestRealmPath() withMigrationBlock:^(RLMMigration *migration, NSUInteger oldSchemaVersion) {
+            XCTAssertEqual(oldSchemaVersion, 0U, @"Initial schema version should be 0");
+
+            XCTAssertTrue([migration deleteTableForClassName:@"DeletedClass"]);
+            XCTAssertFalse([migration deleteTableForClassName:@"NoSuchClass"]);
+
+            XCTAssertThrows([migration deleteTableForClassName:nil]);
+            XCTAssertThrows([migration deleteTableForClassName:StringObject.className]);
+        }];
+        [RLMRealm migrateRealmAtPath:RLMTestRealmPath()];
+    }
+
+    @autoreleasepool {
+        // verify migration
+        RLMRealm *realm = [self realmWithTestPath];
+        XCTAssertFalse(realm.group->has_table(RLMStringDataWithNSString(RLMTableNameForClass(@"DeletedClass"))), @"The deleted class should not have a table.");
+    }
+
+    [RLMRealm setSchemaVersion:0 forRealmAtPath:RLMTestRealmPath() withMigrationBlock:nil];
+    XCTAssertThrows([RLMRealm migrateRealmAtPath:RLMTestRealmPath()]);
```

Comment:
This last assertion doesn't appear to be testing anything related to this change.